twoliter/build.rs

```diff
@@ -0,0 +1,87 @@
+/*!
+
+Prepare and package embedded tools in a tarball to be included with Twoliter.
+
+!*/
+
+// The performance cost of this is infinitesimal, and we get a better panic stack with `expect`.
+#![allow(clippy::expect_fun_call)]
+
+use bytes::BufMut;
+use flate2::write::GzEncoder;
+use flate2::Compression;
+use sha2::{Digest, Sha256};
+use std::path::{Path, PathBuf};
+use std::{env, fs};
+
+const DATA_INPUT_DIR: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/embedded");
+const TOOLS_HASH_RS_PATH: &str = concat!(env!("CARGO_MANIFEST_DIR"), "/src/tools_hash.rs");
+
+fn main() {
+    println!("cargo:rerun-if-changed={}", DATA_INPUT_DIR);
+    let data_input_dir = PathBuf::from(DATA_INPUT_DIR);
+
+    // This is the directory that cargo creates for us so that we can pass things from the build
+    // script to the main compilation phase.
+    let out_dir =
+        PathBuf::from(env::var("OUT_DIR").expect("The cargo variable 'OUT_DIR' is missing"));
+
+    // This is where we will copy all of the things we want to add to our tarball. We will then
+    // compress and tar this directory.
+    let tools_dir = out_dir.join("tools");
+    fs::create_dir_all(&tools_dir).expect(&format!(
+        "Unable to create directory '{}'",
+        tools_dir.display()
+    ));
+
+    // This is the filepath to the tarball we will create.
+    let tar_path = out_dir.join("tools.tar.gz");
+
+    // TODO - use Makefile.toml when we have imported the Bottlerocket monorepo git history.
+    let makefile_source = data_input_dir.join("Makefile.temp.toml");
+    copy_file(makefile_source, tools_dir.join("Makefile.toml"));
+
+    // Create tarball in memory.
+    let mut buf_writer = Vec::new().writer();
+    let enc = GzEncoder::new(&mut buf_writer, Compression::default());
+    let mut tar = tar::Builder::new(enc);
+    tar.append_dir_all("", &tools_dir).unwrap();
+
+    // Drop tar object to ensure any finalizing steps are done.
+    drop(tar);
+
+    // Get a reference to the tarball bytes.
+    let tar_data = buf_writer.get_ref();
```

Comment:
nit: to indicate that it's been compressed already -
```suggestion
    // Get a reference to the tarball bytes.
    let tar_gz_data = buf_writer.get_ref();
```